public/pages/PolicyDetails/containers/PolicyDetails/PolicyDetails.tsx

```diff
@@ -0,0 +1,243 @@
+import DeleteModal from "../../components/DeleteModal/DeleteModal";
+import States from "../../../VisualCreatePolicy/components/States";
+
+interface PolicyDetailsProps extends RouteComponentProps {
+  policyService: PolicyService;
+}
+
+interface PolicyDetailsState {
+  policyId: string;
+  channelId: string;
+  primaryTerm: number;
+  lastUpdated: string;
+  description: string;
+  sequenceNumber: number;
+  schemaVersion: number;
+  ismTemplates: ISMTemplate[];
+
+  isJSONModalOpen: boolean;
+  policyJSON: any;
+  isDeleteModalVisible: boolean;
+}
+
+export default class PolicyDetails extends Component<PolicyDetailsProps,
+  PolicyDetailsState> {
+    static contextType = CoreServicesContext;
+    constructor(props: PolicyDetailsProps) {
+      super(props);
+
+      this.state = {
+        policyId: "",
+        channelId: "",
+        primaryTerm: -1,
+        lastUpdated: "",
+        description: "",
+        sequenceNumber: -1,
+        schemaVersion: -1,
+        ismTemplates: [],
+
+        isJSONModalOpen: false,
+        policyJSON: {policy:{states:[]}},
+        isDeleteModalVisible: false,
+      };
+    }
+
+    componentDidMount = async (): Promise<void> => {
+      this.context.chrome.setBreadcrumbs([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.INDEX_POLICIES]);
+      const { id } = queryString.parse(this.props.location.search);
+      if (typeof id === "string") {
+        this.context.chrome.setBreadcrumbs([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.INDEX_POLICIES, { text: id }]);
+        this.props.history.push(`${ROUTES.POLICY_DETAILS}?id=${id}`);
+        console.log("Getting policy!");
+        await this.getPolicy(id);
+        this.forceUpdate();
+      } else {
+        this.context.notifications.toasts.addDanger(`Invalid policy id: ${id}`);
+        this.props.history.push(ROUTES.INDEX_POLICIES);
+      }
+    };
+
+    getPolicy = async (policyId: string): Promise<void> => {
+      try {
+        const { policyService } = this.props;
+        const response = await policyService.getPolicy(policyId);
+
+        if (response.ok) {
+          console.log("Response: ", response.response);
+          this.setState({
+            policyId: response.response.id,
+            channelId: response.response.policy.error_notification,
+            primaryTerm: response.response.primaryTerm,
+            lastUpdated: response.response.policy.last_updated_time,
+            policyJSON: response.response,
+            description: response.response.policy.description,
+            sequenceNumber: response.response.seqNo,
+            schemaVersion: response.response.policy.schema_version,
+          })
+          if (response.response.policy.ism_template) {
+            this.setState({ismTemplates: response.response.policy.ism_template,});
+          }
+        }
+      } catch (err) {
+        this.context.notifications.toasts.addDanger(getErrorMessage(err, "Somethin' happened"));
```

Comment:
Let's remove "Somethin' happened" and put something else, I'm assuming it's the default message if it can't find one?

Reply:
Yep, this error handling was incomplete. Replaced with a danger toast and redirect to the main page.